pkg/cnb/create_builder.go

```diff
@@ -0,0 +1,74 @@
+package cnb
+
+import (
+	"github.com/google/go-containerregistry/pkg/authn"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+
+	"github.com/pivotal/kpack/pkg/apis/build/v1alpha1"
+	experimentalV1alpha1 "github.com/pivotal/kpack/pkg/apis/experimental/v1alpha1"
+)
+
+type Client interface {
+	Fetch(keychain authn.Keychain, repoName string) (v1.Image, error)
+	Save(keychain authn.Keychain, tag string, image v1.Image) (string, error)
+}
+
+type StoreFactory interface {
+	MakeStore(keychain authn.Keychain, storeImage string) (Store, error)
+}
+
+type Store interface {
+	FetchBuildpack(id, version string) (RemoteBuildpackInfo, error)
+}
+
+type RemoteBuilderCreator struct {
+	RemoteImageClient Client
+	StoreFactory      StoreFactory
```

Comment:
Out of curiosity, what lead to migrating this to use an interface vs a func signature?

Reply:
I just preferred giving it a name rather than leaving it as an anonymous function definition.